site/src/api/api.ts

```diff
@@ -57,1807 +153,1777 @@ export const withDefaultFeatures = (
+        // query result
+        return null;
+      }
+
+      throw error;
+    }
+  };
+}
+
+// This is a hard coded CSRF token/cookie pair for local development. In prod,
+// the GoLang webserver generates a random cookie with a new token for each
+// document request. For local development, we don't use the Go webserver for
+// static files, so this is the 'hack' to make local development work with
+// remote apis. The CSRF cookie for this token is "JXm9hOUdZctWt0ZZGAy9xiS/gxMKYOThdxjjMnMUyn4="
+const csrfToken =
+  "KNKvagCBEHZK7ihe2t7fj6VeJ0UyTDco1yVUJE8N06oNqxLu5Zx1vRxZbgfC0mJJgeGkVjgs08mgPbcWPBkZ1A==";
+
+// Always attach CSRF token to all requests. In puppeteer the document is
+// undefined. In those cases, just do nothing.
+const tokenMetadataElement =
+  typeof document !== "undefined"
+    ? document.head.querySelector('meta[property="csrf-token"]')
+    : null;
+
+interface ClientApi {
+  api: Api;
+  getCsrfToken: () => string;
+  setSessionToken: (token: string) => void;
+  setHost: (host: string | undefined) => void;
+  getAxiosInstance: () => AxiosInstance;
+}
+
+export class Client implements ClientApi {
+  private readonly axios: AxiosInstance;
+  readonly api: Api;
+
+  constructor() {
+    this.axios = this.getConfiguredAxiosInstance();
+    this.api = new Api(this.axios);
+  }
+
+  private getConfiguredAxiosInstance(): AxiosInstance {
+    const axios = globalAxios.create();
+
+    // Adds 304 for the default axios validateStatus function
+    // https://github.com/axios/axios#handling-errors Check status here
+    // https://httpstatusdogs.com/
+    axios.defaults.validateStatus = (status) => {
+      return (status >= 200 && status < 300) || status === 304;
+    };
+
+    const metadataIsAvailable =
+      tokenMetadataElement !== null &&
+      tokenMetadataElement.getAttribute("content") !== null;
+
+    if (metadataIsAvailable) {
+      if (process.env.NODE_ENV === "development") {
+        // Development mode uses a hard-coded CSRF token
+        this.axios.defaults.headers.common["X-CSRF-TOKEN"] = csrfToken;
+        axios.defaults.headers.common["X-CSRF-TOKEN"] = csrfToken;
+        tokenMetadataElement.setAttribute("content", csrfToken);
+      } else {
+        axios.defaults.headers.common["X-CSRF-TOKEN"] =
+          tokenMetadataElement.getAttribute("content") ?? "";
+      }
+    } else {
+      // Do not write error logs if we are in a FE unit test.
+      if (process.env.JEST_WORKER_ID === undefined) {
+        console.error("CSRF token not found");
+      }
     }
+
+    return axios;
   }
-};
+
+  getCsrfToken = (): string => {
+    return csrfToken;
+  };
+
+  setSessionToken = (token: string): void => {
+    this.axios.defaults.headers.common["Coder-Session-Token"] = token;
+  };
+
+  setHost = (host: string | undefined): void => {
+    this.axios.defaults.baseURL = host;
+  };
+
+  getAxiosInstance = (): AxiosInstance => {
+    return this.axios;
+  };
+}
+
+export const client = new Client();
```

Comment:
Just a thought: if we named this `API` we could reduce the diff in a bunch of places. Change the `import`, but the code stays the same.